src/libraries/System.Security.Cryptography/ref/System.Security.Cryptography.cs

```diff
@@ -1853,6 +1853,23 @@ internal MLKemAlgorithm() { }
         public static bool operator ==(System.Security.Cryptography.MLKemAlgorithm? left, System.Security.Cryptography.MLKemAlgorithm? right) { throw null; }
         public static bool operator !=(System.Security.Cryptography.MLKemAlgorithm? left, System.Security.Cryptography.MLKemAlgorithm? right) { throw null; }
     }
+    [System.Diagnostics.CodeAnalysis.ExperimentalAttribute("SYSLIB5006")]
+    [System.Runtime.Versioning.UnsupportedOSPlatformAttribute("android")]
```

Comment:
I know we oscillate between liking and hating these attributes in crypto - but I think they belong here. We previously didn't like them because we were using them to indicate limitations of the platform. This however is not a platform limitation, this is a "`MLKemOpenSsl` is not, and will not, be designed to run on these platforms".

_Why not use SupportedOSPlatform("linux")?_

So we don't mistakenly leave out other unixes maintained by the community, like freebsd or illumos, which as far as I am aware, SOSP does not consider "linux".

Reply:
I don't think we want them on the type.  A library that wants to take an MLKemOpenSsl (for whatever reason) in a method signature shouldn't have to align to these guards.  They belong more on the constructor ("you can't make one of these on android/ios/osx/tvos/windows").  But even that feels like it should be redundant given the current "you can only instantiate me from a SafeEvpPKeyHandle", since the ways of getting one of those should have already had that armor.

Reply:
Yeah, type is wrong. Agree on that.

> since the ways of getting one of those should have already had that armor.

That is not true on macOS. `MLKemOpenSsl` is not going to work on macOS no matter what because we use `MLKemOpenSsl.NotSupported` for that OS target. `SafeEvpPKeyHandle.OpenKeyFromProvider` happily still works on macOS though.

In general, I want to really clear and unambiguously paint this type as "Instantiating this on this platforms is not going to work, even if you get a handle from somewhere" and that is by-design.